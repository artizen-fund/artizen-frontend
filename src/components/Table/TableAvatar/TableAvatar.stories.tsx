import TableAvatar from './'

export default {
  title: 'components/TableAvatar',
  component: TableAvatar,
  argTypes: {},
}

export const TableAvatarComponent = (props: any) => <TableAvatar {...props} />
