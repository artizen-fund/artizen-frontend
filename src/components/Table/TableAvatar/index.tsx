import styled from 'styled-components'
import { rgba } from '@lib'
import { palette, breakpoint } from '@theme'

const TableAvatar = styled.div<{
  profileImage?: any
}>`
  display: inline-block;
  width: 32px;
  height: 32px;
  min-width: 32px;

  background-image: url(${props => props.profileImage || '/assets/glyphs/face/20/outline.svg'});
  background-size: ${props => (props.profileImage ? 'cover' : 'auto')};
  background-position: center center;
  background-repeat: no-repeat;
  border-radius: 9999px;
  border: 2px solid ${rgba(palette.white)};
  @media (prefers-color-scheme: dark) {
    border-color: ${rgba(palette.slate)};
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 44px;
    height: 44px;
    min-width: 44px;
  }
`

export default TableAvatar
