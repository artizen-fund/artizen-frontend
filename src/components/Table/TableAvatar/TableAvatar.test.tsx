import renderer from 'react-test-renderer'
import TableAvatar from './'

describe('TableAvatar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TableAvatar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
