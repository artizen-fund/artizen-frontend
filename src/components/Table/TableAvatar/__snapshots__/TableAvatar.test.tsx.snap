// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableAvatar renders correctly 1`] = `
<div
  className="sc-e636bb41-0 bXJVFn"
/>
`;
